### PR TITLE
tests: Avoid Expect(len())

### PR DESCRIPTION
**Pull request by EdDev**
**What this PR does / why we need it**:

The gomega-idiomatic form `Expect(x).To(HaveLen(n))`
is better than `Expect(len(x)).To(Equal(n))`
because when the expectation fails, the tester sees the value of x,
not just its length, and has more information to debug it.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Review by orelmisan — APPROVED** [review]

File: tests/vm_test.go [hunk 1]
```diff
@@ -1101,7 +1101,7 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 					Expect(err).ToNot(HaveOccurred())
 					Expect(newVM.Spec.RunStrategy).ToNot(BeNil())
 					Expect(*newVM.Spec.RunStrategy).To(Equal(v1.RunStrategyHalted))
-					Expect(len(newVM.Status.StateChangeRequests)).To(Equal(0))
+					Expect(newVM.Status.StateChangeRequests).To(HaveLen(0))
```
Comment (dankenigsberg) [inline tests/vm_test.go:1104]:
```suggestion
					Expect(newVM.Status.StateChangeRequests).To(BeEmpty())
```
is more idiomatic

File: tests/vm_test.go [hunk 2]
```diff
@@ -1280,7 +1280,7 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 					Expect(newVM.Spec.RunStrategy).ToNot(BeNil())
 					Expect(*newVM.Spec.RunStrategy).To(Equal(v1.RunStrategyHalted))
 					By("Ensuring stateChangeRequests list is cleared")
-					Expect(len(newVM.Status.StateChangeRequests)).To(Equal(0))
+					Expect(newVM.Status.StateChangeRequests).To(HaveLen(0))
```
Comment (dankenigsberg) [inline tests/vm_test.go:1283]: ditto

**Review by dankenigsberg — CHANGES_REQUESTED** [review]

File: tests/vm_test.go (hunk 1 above)
Comment (EdDev) [reply tests/vm_test.go:1104]: I prefer to handle it in a different PR, this one focuses on one thing and it is much easier for one to review it.

And BTW, I am not really sure it is valid to do such a replacement with `BeEmpty` blindly, as I believe it should only be done for cases where the type is a slice or map, but not for other things (e.g. a string).

Comment (dankenigsberg) [reply tests/vm_test.go:1104]: Ok. I don't mind waiting with `BeEmpty`

Comment (EdDev) [reply tests/vm_test.go:1104]: Feel free to vote :)

**Review by dankenigsberg — APPROVED** [review]

**Review by phoracek — COMMENTED** [review]
/approve

**Pull request merged**